### PR TITLE
feat!: add new rules/configs/plugins

### PR DESCRIPTION
**Pull request by sztok7**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,36 +146,38 @@ module.exports = {
...
             }
         ],
-        "@typescript-eslint/no-unused-vars": ["error", {
-            "varsIgnorePattern": "^_",
-            "ignoreRestSiblings": true
-        }],
         "@typescript-eslint/explicit-function-return-type": ["error", {
             "allowExpressions": true
         }],
+        "@typescript-eslint/consistent-type-definitions": ["error", "type"],
```
Comment (sztok7) [inline index.js:152]: I think it's a good practice to use either types or interfaces (personally prefer types, but this rule can be changed or turned off if preferences differ or if the  project is a dependency of another that relies on a specific type definition style)

File: index.js [hunk 2]
```diff
@@ -39,36 +146,38 @@ module.exports = {
...
+        "@typescript-eslint/consistent-type-imports": "error",
         "@typescript-eslint/no-require-imports": "error",
-        "import/order": [
```
Comment (sztok7) [inline index.js:50]: removed this, since it'll be handled by `simple-import-sort` :)

File: test/validate.spec.js
Comment (sztok7) [inline test/validate.spec.js:1]: most changes are just code style + added some new eslint error assertions

Comment (sztok7) [inline test/validate.spec.js:1]: most changes are just code style + some new rule error assertions

File: index.js [hunk 3]
```diff
@@ -3,33 +3,140 @@ module.exports = {
...
     parserOptions: {
       sourceType: 'module',
       project: './tsconfig.json'
     },
+    extends: [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/eslint-recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking",
+        "prettier",
+        "airbnb-base",
```
Comment (mpetrunic) [inline index.js:16]: Let's not put this under base rule. You should make new file "react.js" that extends from index.js and adds react specific rules.

**Review by mpetrunic — CHANGES_REQUESTED** [review]

**Pull request closed without merging**